test/Microsoft.AspNetCore.OData.Versioning.ApiExplorer.Tests/AspNet.OData/DefaultModelTypeBuilderTest.cs

```diff
@@ -355,6 +356,29 @@ public void substitute_should_generate_types_for_actions_with_the_same_name_in_d
             employeesActionType.Should().HaveProperty<string>( "project" );
         }
 
+        [Fact]
+        public void substitute_should_resolve_types_that_reference_a_model_that_match_the_edm()
+        {
+            // arrange
+            var modelBuilder = new ODataConventionModelBuilder();
+
+            var shipment = modelBuilder.EntitySet<Shipment>( "Shipments" ).EntityType;
+            shipment.Ignore( s => s.ShippedOn );
+            var originalType = typeof(Shipment);
+
+            modelBuilder.EntitySet<Address>("Addresses");
+            var addressType = typeof(Address);
+
+            var context = NewContext( modelBuilder.GetEdmModel() );
+
+            // act
+            addressType.SubstituteIfNecessary( context );
+            var substitutionType = originalType.SubstituteIfNecessary( context );
+
+            // assert
+            Assert.False( substitutionType is TypeBuilder );
```

Comment:
This isn't wrong, but for consistency let's continue to use Fluent Assertions, which would look like this:

```c#
substitutionType.Should().NotBeOfType( typeof( TypeBuilder ) );
```

>Don't forgot to format the document ;)

Reply:
The problem with this is that it checks for the type that the **TypeBuilder** is building, meaning that the test will succeed even if it is a **TypeBuilder**.

I've tried:
`NotBeOfType<TypeBuilder>();`
`NotBe<TypeBuilder>();`
`NotBeDerrivedFrom<TypeBuilder>();`

With each of these methods the test will succeed even if the type is a **TypeBuilder**.
Maybe there is a method or property that extracts **TypeBuilder** from the type or that indicates it is a **TypeBuilder** that could be used instead, but unless there is a Fluent assertion that works for this I think it's best to leave it as is.

Reply:
Interesting... could be a bug in Fluent Assertions. The `is` keyword is basically the same thing as `Type.IsAssignableFrom`. Correctness is more important here. I'll consider revisiting this some time in the future the next time I update Fluent Assertions. Thanks.